docs/docs/APIGuide/Optimizers/Optimizer.md

```diff
@@ -136,6 +154,41 @@ Function setModel will set a new model to the optimizer.
 `newModel`: a model will replace the old model in optimizer.  
  <br>
 ```scala
+optimizer.setTrainData(sampleRDD: RDD[Sample[T]],
+                 batchSize: Int)
+
+optimizer.setTrainData(sampleRDD: RDD[Sample[T]],
+                 batchSize: Int,
+                 miniBatch: MiniBatch[T])
+
+optimizer.setTrainData(sampleRDD: RDD[Sample[T]],
+
+                 batchSize: Int,
+                 featurePaddingParam: PaddingParam[T],
+                 labelPaddingParam: PaddingParam[T])
+
+optimizer.setTrainData(sampleRDD: RDD[Sample[T]],
+                        batchSize: Int,
+                        featurePaddingParam: PaddingParam[T])
+```
+the overloaded set of methods `setTrainData` allows user to replace the trainig data before calling optimize(). The mearning of arguments are the same as in the Factory methods:
```

Comment:
mearning -> meaning

Reply:
can we replace the training data after calling optimize?

Reply:
why we need so ? it would be used like setTrainData, optimize(), setTrainData, optimize()